dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,442 @@
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true),
+                "param2", Param.newParam("true", false, Type.BOOL, "label", "hint", true),
+                "param3", Param.newParam("val-2", true, Type.STRING, "label", "hint", true),
+                "param4", Param.newParam("true", true, Type.BOOL, "label", "hint", true)
+        );
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d", System.currentTimeMillis());
+        final String fileName = String.format("%s.yml", serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, serviceKey,
+                "Test-hidden-secret-protection", false, initialParamsMap);
+
+        final Response serviceIntegrationResponse = serviceIntegrationResource
+                .createServiceIntegration(request, response,
+                        createFormDataMultiPart(fileName, inputStream));
+        Assert.assertNotNull(serviceIntegrationResponse);
+        Assert.assertEquals(HttpStatus.SC_OK, serviceIntegrationResponse.getStatus());
+
+        final List<String> sites = new ArrayList<>();
+
+        final int max = 10;
+        for(int i = 1; i <= max; i++) {
+            sites.add(
+               createServiceIntegrationParams(initialParamsMap, serviceKey, request, response).getIdentifier()
+            );
+        }
+
+        //The Service does exist and so it does the secrets.
+        final Response hostIntegrationsResponse = serviceIntegrationResource.getServiceIntegrationByKey(request, response, serviceKey);
+        Assert.assertEquals(HttpStatus.SC_OK, hostIntegrationsResponse.getStatus());
+        final ResponseEntityView responseEntityView = (ResponseEntityView) hostIntegrationsResponse.getEntity();
+        final ServiceIntegrationSiteView serviceIntegrationHostView = (ServiceIntegrationSiteView) responseEntityView.getEntity();
+        final ServiceIntegrationView serviceIntegrationView = serviceIntegrationHostView.getService();
+        Assert.assertEquals(max, serviceIntegrationView.getConfigurationsCount());
+
+        for (final String siteId : sites){
+            final Response detailedIntegrationResponse = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, siteId);
+            Assert.assertEquals(HttpStatus.SC_OK, detailedIntegrationResponse.getStatus());
+            final ResponseEntityView responseEntityView3 = (ResponseEntityView) detailedIntegrationResponse.getEntity();
+            final ServiceIntegrationDetailedView serviceIntegrationDetailedView = (ServiceIntegrationDetailedView) responseEntityView3.getEntity();
+            final Map<String,Secret> secrets = serviceIntegrationDetailedView.getSecrets();
+            for (Entry<String, Secret> secretEntry : secrets.entrySet()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'secretEntry' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)